src/actix/mod.rs

```diff
@@ -66,6 +69,12 @@ pub fn init(
 
             App::new()
                 .wrap(Compress::default()) // Reads the `Accept-Encoding` header to negotiate which compression codec to use.
+                // api_key middleware
+                // note: the last call to `wrap()` or `wrap_fn()` is executed first
+                .wrap(Condition::new(
```

Comment:
Maybe we can enable this as first check? That would leave malicious users out early. Or do you think this is better?

Reply:
I like where I put this. (Though I'm always happy to be told *why* I'm wrong.)

If I put this before logging and telemetry then you lose visibility into unauthorized requests. (They don't make it to logging or telemetry.)

With CORS I was thinking that was a higher level check, so it should go first. I suppose they are both HTTP headers though.

Reply:
tbh, I see no problem with this place as is. @timvisee are there any concerns?

Reply:
No concerns. Marking as resolved.